scrapy/core/downloader/handlers/http11.py

```diff
@@ -440,6 +441,14 @@ def _cb_bodydone(self, result, request, url):
             certificate=result["certificate"],
             ip_address=result["ip_address"],
         )
+        if result.get("failure"):
+            # This failure is not the same object that will reach the errback,
+            # so we need to temporarily store the response in the exception.
+            # It will be moved to the failure in core/scraper.py
+            failure = result["failure"]
+            failure.value.response = response
+            raise failure
```

Comment:
hm, I was not aware raising a Failure instances directly is what people do.. Could you please double-check that it doesn't cause any issues, as compared to e.g. calling a method on failure to raise an exception? Or maybe it is possible to return a failure instead of raising it? Sorry, I'm clueless :)

Reply:
I _believe_ it is the same thing, but I'm not 100% sure so don't quote me on that. Tests are not affected by this change, but in any case, let's play it safe and return instead. I hoped it would have been the key to resolve https://github.com/scrapy/scrapy/pull/4559#discussion_r425334750, but sadly it is not.